.github/workflows/unit-test.yml

```diff
@@ -0,0 +1,79 @@
+name: Unit and linting tests
+
+on:
+  workflow_call:
+    inputs:
+      plugin:
+        type: string
+jobs:
+  unit-tests:
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Set up Python ${{matrix.python-version}}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{matrix.python-version}}
+          cache: 'pip'
+      - name: Install dependencies
+        run: |
+          cd ${{ inputs.plugin }}
+          pip install git+https://github.com/kedro-org/kedro@main
+          pip install -r test_requirements.txt
+          pip freeze
+      - name: Run unit tests
+        run: make plugin=${{ inputs.plugin }} test
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Set up Python 3.8
```

Comment:
Nit: Any reason not to lint on the latest supported version instead of 3.8?

Reply:
Current setup also runs lint on 3.8. The lint tests fail for other versions of python for `kedro-datasets` because of [this](https://github.com/kedro-org/kedro-plugins/blob/dd42e620efa4ca82ec31c8403cf617e4082dba4f/kedro-datasets/test_requirements.txt#L54) `snowflake-snowpark-python` is only installed for python version 3.8 so pylint throws import errors. We can change the python version to 3.10 and suppress the error

Reply:
Thanks for answering, I'm fine with either route (or doing this in a separate task in the future); I was just wondering. :)